### DISCUSSION TITLE
Kernel dies importing spacy

### DISCUSSION THREAD

**Discussion by bosanipietro** [explosion/spaCy#7909] [Help: Installation] [answered]
## How to reproduce the behaviour

`import spacy`

## Your Environment

* Operating System: macOS with Apple M1 chip
* Python Version Used: 3.9.4
* spaCy Version Used: 3.0.0
* Environment Information:

Importing spaCy library into Jupiter notebook within the above environment kills my kernel, with the package failing to import. Any help would be greatly appreciated, thanks for all your hard work!

**Comment by polm** [comment, marked as answer]
Sorry you're having trouble with that. We have had reports of issues with M1 chips, and in general Python and pip had trouble adjusting immediately after they were released, though at this point I think most people are able to use it without issue. 

1. Can you upgrade spaCy to the most recent version and try again?
2. Do you not get any error output? If you do can you share it? How do you know your kernel died?
3. Can you import spaCy in Python on the command line?

**Reply by bosanipietro** [reply]
I tried with version 3.0.6 and still get the same issue. 
I don't get any error output. Trying to import it into Jupiter Notebook, I get a message that the kernel has dies (as shown in image below) in the upper right corner and the block does not return anything and appears not to have been executed (as shown by the square with no number next to it). 
<img width="1409" alt="Screenshot 2021-04-27 at 09 53 13" src="https://user-images.githubusercontent.com/51325989/116206096-be73c080-a73e-11eb-9bca-385644a70eaa.png">

When I try to import it in Python command line I get the following message: `Illegal instruction: 4`

**Reply by polm** [reply]
> Illegal instruction: 4

Ah, that helps. It looks like the issue is probably that your Rosetta setting is wrong - you're either not using it when you should or using it when you shouldn't. 

Not sure how you're starting your notebook, but adjusting your Rosetta settings. I think you can right click on the app, click "get info", and choose to use Rosetta or not.

**Reply by sakaguchikenta** [reply]
could u resolve this problem?
i have same problem now.
my pc is m1 too

**Reply by polm** [reply]
@sakaguchikenta Did you try the steps suggested above?

**Reply by abhi-vellala** [reply]
@polm You mean which app should I choose to use Rosetta? Anaconda?  I've tried to reinstall spacy. It still crashes the kernel.

**Reply by polm** [reply]
@abhi-vellala If you are getting `Illegal instruction: 4`, the issue is that your Python and the package you're trying to use have different CPU architectures. You might be able to fix that in Rosetta settings as indicated above. If that doesn't work, please answer the questions above so we have information to help you - just knowing the kernels crashes doesn't give us enough information to figure out what's wrong.

**Reply by isobarbaric** [reply]
similar issue, I couldn't get ``python -m spacy download en_core_web_sm`` to work normally, so I tried with Rosetta, but to no avail; the error I keep getting is this: ``[1] (a random number here) illegal hardware instruction  python -m spacy download en_core_web_sm`` (the number after the ``[1]`` is different every time)

**Reply by rohitsahayy** [reply]
i am having the same problem . The kernel dies when trying to import Spacy. 
Operating System: macOS with Apple M2 chip
Python Version Used: 3.9.12
spaCy Version Used: 3.5.3

**Reply by svlandeg** [reply]
Please open a new issue if you run into this, and share all information as outlined in the section "How to ask a good question" [here](https://github.com/explosion/spaCy/discussions/8226). Thanks!

**Comment by wentingd17** [comment]
I had the same issue before.
I tried to create a conda Env, install spacy packages there and add it to my jupyter notebook. Then run the jupyter notebook in that new conda Env then it works.

**Reply by abhi413sh** [reply]
@wentingd17  how you did it?

**Comment by bobiq11** [comment]
I also have this problem. How do I solve this?

**Reply by svlandeg** [reply]
Can you answer the questions Paul summed up [here](https://github.com/explosion/spaCy/discussions/7909#discussioncomment-662822)?

**Comment by anammari** [comment]
I am encountering the same issue on a Jupyter server deployed as a HuggingFace Space. 
It has been working with no issues previously but just today started facing the kernel dies issue

<img width="565" alt="image" src="https://github.com/explosion/spaCy/assets/2264734/6acf60e2-b2b3-4894-afbe-2a2749565218">

Tried the latest SpaCy version v3.5.3 as well as v3.5.0

Python version is v3.10

Running "!pip show spacy" in Jupyter works fine and produces the following output:
Name: spacy
Version: 3.5.0
Summary: Industrial-strength Natural Language Processing (NLP) in Python
Home-page: https://spacy.io/
Author: Explosion
Author-email: contact@explosion.ai
License: MIT
Location: /home/jovyan/.local/lib/python3.10/site-packages
Requires: catalogue, cymem, jinja2, langcodes, murmurhash, numpy, packaging, pathy, preshed, pydantic, requests, setuptools, smart-open, spacy-legacy, spacy-loggers, srsly, thinc, tqdm, typer, wasabi
Required-by:

**Reply by vinbo8** [reply]
Could you open a new thread for this issue, with server logs instead of images if you have access to those?

**Reply by anammari** [reply]
Thanks @vin-ivar 
New issue has been logged https://github.com/explosion/spaCy/issues/12665
I've added both the HF Space build logs and the Space container run-time logs

**Comment by Aisling-Kinsella** [comment]
I'm having this same problem when I import spacy into an environment pre-loaded with top2vec. Previous spacy env working fine, top2vec env working fine before spacy install in that environment. Can't find a workaround except to switch environments for relevant work. Is there some way to find out what is causing this problem though, as I've followed the flow in answer above and spacy is updated, installed also via command line into the env but there's no problem with the install, just the dead kernel when the code hits the import spacy command in jupyter notebook

**Reply by adrianeboyd** [reply]
This is probably due to an interaction with `tensorflow`. If you uninstall tensorflow, does spacy work again?

`spacy` (well, `thinc`) imports tensorflow automatically if it's installed, but users rarely use tensorflow with spacy. For now, the easiest workaround is to use a separate venv for spacy where tensorflow isn't installed.

But this is becoming more of a problem with newer versions of tensorflow, so I think we should modify `thinc` so that it no longer imports tensorflow by default, only if the user requests it: https://github.com/explosion/thinc/pull/890

**Comment by patkovacs** [comment]
if i install it with conda, it works:
conda install -c conda-forge spacy
